### DISCUSSION TITLE
Support 2-variable charts to Study View

### DISCUSSION THREAD

**Discussion by jjgao** [cBioPortal/cbioportal#8828] [Ideas]
The Study View currently supports only one two-variable chart: Mutations vs FGA. It would be useful to plot a chart of any two variables. Much of the functionality already exists in Plots tab of Results view and can be integrated into Study View (actually, there have been comments about Study View being a more natural place for Plots tab).

Here is some food for thoughts on the user interface/workflow.
- Entry point can be another tab (2-variable Plots?) in the Charts menu
- User will be able to select any clinical data or genomic data (similar to Plots?)
- The new Chart will be displayed in the Study View, which will allow interaction with other charts, ie. showing only the selected cohort, and allowing selection of samples/patients in the plot
- Each plot will have a menu that allow reconfigure the plot (similar to Plots tab?)

**Comment by adamabeshouse** [comment]
Are we bringing all the settings over from plots tab, including options to swap, show mutation and cna genomic coloring, filter categories, searching cases and mutations to highlight samples, show p-value, show regression line?

**Reply by jjgao** [reply]
@adamabeshouse I think we should bring as much setting as possible...

**Comment by adamabeshouse** [comment]
what is the expected size for these charts? same size as the other charts, or extra large?

**Reply by jjgao** [reply]
Most can probably fit into a 2x2 grid (same as Mut vs CNA), but some may require different configuration when an axis is long.

**Comment by adamabeshouse** [comment]
Brainstorming session summary @alisman @tmazor @jjgao @inodb 

Proposals:
1) Split configuration into 2 phases: add chart menu contains data selection, then mouseover menu contains fine detail changing, such as log scale, p-value, swap axes, regression line, search, etc.

2) Bring up a modal for configuration that contains all the settings

3) Bring up a modal for configuration that contains the entire plots tab, so there is a chart preview.

Questions remaining:

- Is it useful enough to justify the work?
- How much work will it be?

Next steps:

- Adam will implement (3), at least as a prototype. Replace gene selection dropdown with user-input gene validation box.

**Reply by adamabeshouse** [reply]
Temporary tab name: X vs Y

**Comment by jjgao** [comment]
One more thing to consider - there is a key difference between results view and study view. Study view does not have sample-sample data (only summary data) for performance reason (?).

**Reply by adamabeshouse** [reply]
its not a big deal, we can just use the results view APIs

**Comment by jjgao** [comment]
As discussed on [slack](https://cbioportal.slack.com/archives/C359NNB41/p1629750973201500), we decided to make it simple and focus on expanding Mut vs CNA chart to any numerical clinical attributes as the first step. We’ll then learn and improve.

**Reply by adamabeshouse** [reply]
Epic here: https://github.com/cBioPortal/cbioportal/issues/8866
